### ISSUE TITLE
False positive check leads to compilation fail

### ISSUE THREAD

**Issue by denis-bezrukov** [square/anvil#187]
First of all thanks for quick support for dagger's AssistedInject! 

We did a migration to it with `2.0.12-SNAPSHOT` anvil version and everything worked fine, but with update to stable 2.0.12 compilation started failing with `@Provides methods cannot be abstract` error. 

Module failed to compile looks like this: 
```
@Module
@ContributesTo(AppComponent::class)
interface Module {

    @Binds
    @Singleton
    fun bindsDependency(i: DependencyImpl): Dependency

    companion object {
        @Provides
        @Singleton
        fun providesDepB(depA: DepA): DepB = DepB(depA)
    }
}
```
This code is valid with both dagger kapt and anvil 2.0.11. 

It looks like it was introduced in #186  https://github.com/square/anvil/pull/186/files#diff-c59935dc466bb8ab8c74b8e866de744af48eeddaa9340add1833bb09776ff354R63
and `@Provides` function should be filtered if it's inside companion object

**Comment by vRallev** [comment]
Yes, that's a last minute regression. Thanks for the sample, that's very useful!

**Comment by vRallev** [comment]
@denis-bezrukov I've published a snapshot release with the fix `2.0.13-SNAPSHOT`. Could you please confirm that the issue is gone before I push a new release? Thanks!

**Comment by denis-bezrukov** [comment]
Sorry for delay, works fine on both 2.0.14 and 2.1.0, thanks!
